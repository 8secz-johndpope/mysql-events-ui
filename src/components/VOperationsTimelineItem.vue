<script>
export default {
  name: 'OperationsTimelineItem',
  props: {
    operation: {
      type: Object,
      required: true,
    }
  },
}
</script>

<template>
  <div class="timeline-item">
    <div class="timeline-item__header">
      <div class="type">
        {{ operation.type }}
      </div>

      <div class="date">
        <timeago
          :since="operation.timestamp"
          :auto-update="60"
        ></timeago>
      </div>
    </div>
    <div class="timeline-item__content">
      <div class="schema">
        {{ operation.schema }}
      </div>

      <div class="table">
        {{ operation.table }}
      </div>

      <button class="more-information">
        <i class="material-icons">
          unfold_more
        </i>
      </button>
    </div>
  </div>
</template>

<style scoped>
.timeline-item {
  position: relative;
  box-shadow: 0 1px 4px rgba(0, 0, 0, 0.06), 0 3px 6px rgba(0, 0, 0, 0.26);
}

.timeline-item__header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 4px 10px;
  background-color: #D11345;
  color: #FFF;
}

.timeline-item__content {
  display: grid;
  grid-template-columns: 1fr auto;
  grid-template-rows: 1fr 1fr;
  justify-items: start;
  padding: 4px 0 4px 10px;
}

.schema {
  grid-column: 1;
  grid-row: 1;
}

.table {
  grid-column: 1;
  grid-row: 2;
}

.more-information {
  grid-column: 2;
  grid-row: 1 / 3;
  justify-self: end;
  align-self: center;
  background-color: transparent;
  outline: none;
  border: none;
  cursor: pointer;
  border-radius: 2px;
  transition: .2s background-color;
}

.more-information > i {
  transition: .2s color;
}

.more-information:hover > i {
  color: #247BA0;
}

.type {
  font-weight: 500;
}

.date {
  font-size: 12px;
  font-weight: 500;
}
</style>

