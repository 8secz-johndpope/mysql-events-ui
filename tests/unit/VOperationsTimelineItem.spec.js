import { mount, shallowMount, createLocalVue } from '@vue/test-utils'
import toNow from 'date-fns/distance_in_words_to_now'
import VueTimeago from 'vue-timeago'
import VOperationsTimelineItem from '@/components/VOperationsTimelineItem'
import socketFixture from './fixtures/socketServer'

const localVue = createLocalVue()
localVue.use(VueTimeago)

describe('TheOperationsLog', () => {
  let props

  const build = () => {
    const options = {
      propsData: props,
      localVue,
    }

    const wrapper = shallowMount(VOperationsTimelineItem, options)
    const wrapperMounted = mount(VOperationsTimelineItem, options)

    return {
      wrapper,
      type: () => wrapper.find('.type'),
      content: () => wrapper.find('.timeline-item__content'),
      date: () => wrapperMounted.find('.date'),
    }
  }

  beforeEach(() => {
    props = {
      operation: {}
    }
  })

  it('renders the component correctly', () => {
    // arranje
    const { wrapper } = build()

    // assert
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders operation information', () => {
    // arranje
    props.operation = socketFixture.response.INSERT
    const { type, content, date } = build()

    // assert
    expect(type().exists()).toBe(true)
    expect(type().text()).toContain(props.operation.type)

    expect(content().exists()).toBe(true)
    expect(content().text()).toContain(props.operation.schema)

    expect(content().exists()).toBe(true)
    expect(content().text()).toContain(props.operation.table)

    expect(date().exists()).toBe(true)
    expect(date().text()).toContain(toNow(props.operation.timestamp))
  })
})