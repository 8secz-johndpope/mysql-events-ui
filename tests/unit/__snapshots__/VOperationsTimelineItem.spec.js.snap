// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheOperationsLog renders the component correctly 1`] = `
<div class="timeline-item">
  <div class="timeline-item__header">
    <div class="type">

    </div>
    <div class="date">
      <vuecomponent-stub auto-update="60"></vuecomponent-stub>
    </div>
  </div>
  <div class="timeline-item__content">
    <div class="schema">

    </div>
    <div class="table">

    </div>
    <button class="more-information">
      <i class="material-icons">
        unfold_more
      </i>
    </button>
  </div>
</div>
`;
